### PR TITLE
Add Scanorama integration to external API

### PR DESCRIPTION
**Pull request by brianhie**
Hello! I thought it might be useful to add [Scanorama ](https://github.com/brianhie/scanorama) to the set of integration methods in scanpy.external.pp.

I've followed the example laid out in https://github.com/theislab/scanpy/pull/1306 pretty closely. I've also done some small scale integration testing on a local machine just to verify that batch effect correction indeed works when called with this API.

Let me know what I can do to help get this merged!

### CODE REVIEW

**Review by ivirshup — COMMENTED** [review]
Looks good to me! Thanks for the contribution.

Could you just add a point to the release notes about this?

**Comment by LuckyMD** [comment]
Hi @brianhie,

It's great that you're contributing to Scanpy to make the interoperability even easier (I guess it was already quite good given you built on `AnnData`). We have been evaluating data integration methods and in which Scanorama performed quite well (you may have seen the [preprint](https://www.biorxiv.org/content/10.1101/2020.05.22.111161v2)). One aspect that would make it even easier to use the tool that we were missing in the comparison is a small tutorial. The example in the function docstring is already very helpful, but do you think it would be possible to add a quite jupyter notebook in this direction? This is obviously a request outside of this PR.

On the topic of the PR, I wonder if `adata.obsm['X_pca_scanorama']` is a good default name for the generated embedding, and not just `adata.obsm['X_scanorama']` as the standard user may not have delved into the methodology as much.

File: scanpy/tests/external/test_scanorama_integrate.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+def test_scanorama_integrate():
+    """
+    Test that Scanorama integration works.
+
+    This is a very simple test that just checks to see if the Scanorama
+    integrate wrapper succesfully added a new field to ``adata.obsm``
+    and makes sure it has the same dimensions as the original PCA table.
+    """
+    adata = sc.datasets.pbmc3k()
```
Comment (falexwolf) [inline scanpy/tests/external/test_scanorama_integrate.py]: Taking the `pbmc68k_reduced()` dataset, here, would hopefully speed up tests. Don't you think @ivirshup?

We'll run into infinite test times at some point if every tool takes several seconds to run.

**Comment by falexwolf** [comment]
Great! Thanks, @brianhie!

**Comment by brianhie** [comment]
Awesome, thanks everyone.

@ivirshup I added something to the release notes in the latest commit. I hope the formatting is okay -- let me know if there's some better way to do it.

@LuckyMD I've seen your benchmarking preprint and admire the work! For the current API, I'm currently mooching off of tutorials made by others: one which is simpler and one (included in the scanpy tutorials) that is a little more advanced: https://github.com/brianhie/scanorama#full-tutorial. Should this get merged and included in the scanpy API, I promise I'll make a new notebook-based tutorial (probably in Google Colab) that shows off the new API and include a link to it from the Scanorama GitHub README.md. I also agree with shortening the default embedding to `'X_scanorama'` and have done that in the latest commit.

@falexwolf Happy to make any changes to the tests if you think that will boost performance, if you'd like.

**Comment by LuckyMD** [comment]
Hah, I like the community use here! Looking forward to your tutorial then!

File: scanpy/tests/external/test_scanorama_integrate.py (hunk 1 above)
Comment (ivirshup) [reply scanpy/tests/external/test_scanorama_integrate.py]: Good point. At the moment I've been more concerned about the install time of all the dependencies of external modules, but it's good to cut down everywhere we can.

Comment (brianhie) [reply scanpy/tests/external/test_scanorama_integrate.py]: Made the changes accordingly!

**Comment by brianhie** [comment]
@ivirshup thanks for the merge!

**Comment by ivirshup** [comment]
Thanks for the contribution! Sorry about the wait on releasing this!

**Comment by hurleyLi** [comment]
Hi @ivirshup , I'm wondering if you plan to implement the `scanorama_correct` method as well in addition to the `scanorama_integrate` method. #2002 

Thanks!

**Pull request merged**